codespell_lib/_codespell.py

```diff
@@ -984,6 +985,22 @@ def parse_file(
     return bad_count
 
 
+def flatten_comma_separated_arguments(
+    arguments: List[str],
+) -> List[str]:
+    """
+    >>> flatten_comma_separated_arguments([])
+    []
+    >>> flatten_comma_separated_arguments(None)
+    None
```

Comment:
The type hint on this function promises to return a `list`.

Reply:
Yes, that's disturbing. Does this mean that these type hints are useless?

Reply:
Python is a dynamic / weakly typed language so type hints are just hints.  They have _no effect at runtime_.  They are designed to help tools like `mypy` to find type inconsistencies and bugs.  Using type hints is strongly recommended but they do not make Python a strongly typed language like Go, Rust, or TypeScript.

> __Note__  The Python runtime does not enforce function and variable type annotations. They can be used by third-party tools such as type checkers, IDEs, linters, etc.

https://docs.python.org/3/library/typing.html

Reply:
Yes, I understand that. However, we do run mypy in a CI job. If mypy is unable to detect errors, what's the point of these type hints?